dashboard/app/models/aichat_session.rb

```diff
@@ -0,0 +1,20 @@
+# == Schema Information
+#
+# Table name: aichat_sessions
+#
+#  id                   :bigint           not null, primary key
+#  user_id              :integer
+#  level_id             :integer
+#  script_id            :integer
+#  project_id           :integer
+#  model_customizations :text(65535)
```

Comment:
If we're on the fence between text & json, would it be easier to start with json and change from json->text later vs. the other way around, since the json type enforces valid json?

Reply:
[tl;dr changed to json type columns...a few tangential thoughts below]

hmm yeah that makes sense...for our purposes, I'm not sure if makes much of a difference. Was reading thru docs here, seems like there are some storage benefits, and there are handy additions when writing raw SQL that are possible when the column type is JSON:

https://dev.mysql.com/doc/refman/5.7/en/json.html

Tangentially, I stumbled across this bit of guidance about foreign keys when I was thinking about whether to include foreign keys here or not. It kind of made me wonder if enforcing valid JSON at the database level is considered good or bad by Rails ppl:

https://guides.rubyonrails.org/active_record_migrations.html#active-record-and-referential-integrity

> The Active Record way claims that intelligence belongs in your models, not in the database. As such, features such as triggers or constraints, which push some of that intelligence back into the database, are not recommended.

I don't see any explicit guidance on JSON vs text columns elsewhere though. And databases obviously already enforce types to some degree (strings vs ints etc), so sort of a gray area.

Reply:
@sanchitmalhotra126 you mentioned just model customizations in JSON in the comment below, any objection to messages as well? Per what's coming into the controller right now, I think the messages will also be an array of objects and would be fine to serialize as JSON

> _Makes sense to me - maybe json would be useful for the model_customizations field? But yeah text works just fine since I don't think we'll be doing any complex look ups with that field specifically._

Reply:
FWIW, there is a precedent in the codebase 
![image](https://github.com/code-dot-org/code-dot-org/assets/24951544/e2011c84-016b-4160-81f6-9d5540dbbc2c)

(I think you can tell which camp I'm in based on the search results 😁 )

Reply:
haha you're a pioneer! The contact rollups examples do actually write raw SQL against the JSON field, so the benefits are clear there. I think the JSON type only came out in MySQL 5.7 (our current version), so I'm guessing we'll see more usages as time goes on

https://github.com/code-dot-org/code-dot-org/blob/90fc57339095d6f3588836e299e2f00b1347a149/dashboard/app/models/contact_rollups_pardot_memory.rb#L253-L258